crates/bevy_ecs/src/world/mod.rs

```diff
@@ -2377,6 +2377,255 @@ impl World {
         }
     }
 
+    /// For a given batch of ([`Entity`], [`Bundle`]) pairs,
+    /// adds the `Bundle` of components to each `Entity`.
+    /// This is faster than doing equivalent operations one-by-one.
+    ///
+    /// This will overwrite any previous values of components shared by the `Bundle`.
+    /// See [`World::insert_batch_if_new`] to keep the old values instead.
+    ///
+    /// # Panics
+    ///
+    /// This function will panic if any of the associated entities do not exist.
+    ///
+    /// For the non-panicking version, see [`World::try_insert_batch`].
+    #[track_caller]
+    pub fn insert_batch<I, B>(&mut self, iter: I)
+    where
+        I: IntoIterator,
+        I::IntoIter: Iterator<Item = (Entity, B)>,
+        B: Bundle,
+    {
+        self.insert_batch_with_caller(
+            iter,
+            InsertMode::Replace,
+            #[cfg(feature = "track_change_detection")]
+            Location::caller(),
+        );
+    }
+
+    /// For a given batch of ([`Entity`], [`Bundle`]) pairs,
+    /// adds the `Bundle` of components to each `Entity` without overwriting.
+    /// This is faster than doing equivalent operations one-by-one.
+    ///
+    /// This is the same as [`World::insert_batch`], but in case of duplicate
+    /// components will leave the old values instead of replacing them with new
```

Comment:
```suggestion
    /// components it will leave the old values instead of replacing them with new
```

(Nit) I think adding "it" to the sentence makes it flow better, but I don't feel strongly on it.